### PR TITLE
refactor numanode assert in tests

### PR DESCRIPTION
**Pull request by lplewa**
### Checklist

- [ ] Code compiles without errors locally
- [ ] All tests pass locally
- [ ] CI workflows execute properly

- [ ] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [ ] New tests added, especially if they will fail without my changes
- [ ] Added/extended example(s) to cover this functionality
- [ ] Extended the README/documentation
- [ ] All newly added source files have a license
- [ ] All newly added source files are referenced in CMake files
- [ ] Logger (with debug/info/... messages) is used

### CODE REVIEW

File: test/common/numa_helpers.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+// returns the node where page starting at 'ptr' resides
+static inline void getNumaNodeByPtr(void *ptr, int *node) {
+    int nodeId;
+    int ret =
+        get_mempolicy(&nodeId, nullptr, 0, ptr, MPOL_F_ADDR | MPOL_F_NODE);
+
+    ASSERT_EQ(ret, 0) << "get_mempolicy failed";
+    ASSERT_GE(nodeId, 0)
+        << "get_mempolicy returned - nodeId < 0 - should never happen";
```
Comment (lukaszstolarczuk) [inline test/common/numa_helpers.hpp:24]: returned `- nodeId < 0`? I think I know what you meant, but I believe you can remove the first hypen in this sentence

File: test/common/numa_helpers.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#include <gtest/gtest.h>
+#include <numa.h>
+#include <numaif.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "test_helpers.h"
+
+// returns the node where page starting at 'ptr' resides
+static inline void getNumaNodeByPtr(void *ptr, int *node) {
```
Comment (lukaszstolarczuk) [inline test/common/numa_helpers.hpp:17]: I think you could describe in commit message (for future reference) why you changed this function to return void

File: test/common/numa_helpers.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        << "get_mempolicy returned - nodeId < 0 - should never happen";
+
+    *node = nodeId;
+}
+
+static inline void _assertNode(void *ptr, int nodeId, bool fatal) {
+    int node = -1;
+
+    getNumaNodeByPtr(ptr, &node);
+    if (testing::Test::HasFatalFailure()) {
```
Comment (lukaszstolarczuk) [inline test/common/numa_helpers.hpp:33]: can't you use `ASSERT_NO_FATAL_FAILURE` here...?

File: test/common/numa_helpers.hpp [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        return;
+    }
+    if (fatal) {
+        ASSERT_EQ(nodeId, node);
+    } else {
+        EXPECT_EQ(nodeId, node);
+    }
+}
+
+#define ASSERT_NODE_EQ(ptr, nodeId)                                            \
```
Comment (lukaszstolarczuk) [inline test/common/numa_helpers.hpp:43]: maybe a short description for these defines...?

e.g. `Assert that given nodeId is equal to the node where given ptr resides`

File: test/common/numa_helpers.hpp (hunk 3 above)
Comment (lplewa) [reply test/common/numa_helpers.hpp:33]: Done

File: test/common/numa_helpers.hpp (hunk 1 above)
Comment (lplewa) [reply test/common/numa_helpers.hpp:24]: Done

File: test/common/numa_helpers.hpp (hunk 4 above)
Comment (lplewa) [reply test/common/numa_helpers.hpp:44]: Done

File: test/common/numa_helpers.hpp (hunk 2 above)
Comment (lplewa) [reply test/common/numa_helpers.hpp:17]: Added link to gtest documentation

File: test/common/numa_helpers.hpp [hunk 5]
```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2024 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef UMF_NUMA_HELPERS_HPP
```
[the remark this answers is not in the document]
Comment (lplewa) [reply test/common/numa_helpers.hpp]: done

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**